From 68a60c62f85ffb33a54ad7fcad9f7ca019f37152 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Tue, 29 Mar 2011 20:48:12 -0400
Subject: [PATCH 129/486] ENGR00141135 Camera: All green for the captured
 picture

The root cause is that the waite deq number is not set correctly.

Signed-off-by: Yaoming Qin<B10945@freescale.com>
---
 libcamera/V4l2CapDeviceBase.cpp | 4 ++--
 libcamera/V4l2CsiDevice.cpp     | 9 ++++-----
 2 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/libcamera/V4l2CapDeviceBase.cpp b/libcamera/V4l2CapDeviceBase.cpp
index 1d82f15..d396b98 100755
--- a/libcamera/V4l2CapDeviceBase.cpp
+++ b/libcamera/V4l2CapDeviceBase.cpp
@@ -454,7 +454,7 @@ namespace android{
                 CAMERA_HAL_ERR("VIDIOC_QUERYBUF error\n");
                 return CAPTURE_DEVICE_ERR_SYS_CALL;
             } else {
-                CAMERA_HAL_ERR("VIDIOC_QUERYBUF ok\n");
+                CAMERA_HAL_LOG_RUNTIME("VIDIOC_QUERYBUF ok\n");
             }
 
             mCaptureBuffers[i].length = DevBufQue[i].length= buf.length;
@@ -502,7 +502,7 @@ namespace android{
             CAMERA_HAL_ERR("VIDIOC_STREAMON error\n");
             return CAPTURE_DEVICE_ERR_SYS_CALL;
         } else
-            CAMERA_HAL_ERR("VIDIOC_STREAMON ok\n");
+            CAMERA_HAL_LOG_RUNTIME("VIDIOC_STREAMON ok\n");
         return CAPTURE_DEVICE_ERR_NONE;
     }
 
diff --git a/libcamera/V4l2CsiDevice.cpp b/libcamera/V4l2CsiDevice.cpp
index 68fa091..67041c9 100755
--- a/libcamera/V4l2CsiDevice.cpp
+++ b/libcamera/V4l2CsiDevice.cpp
@@ -140,7 +140,7 @@ namespace android{
         }else{
             //hardcode here for ov3640
             if (strstr(mInitalDeviceName, "3640") != NULL){
-                LOGD("the sensor  is  mInitalDeviceName");
+                CAMERA_HAL_LOG_INFO("the sensor  is  mInitalDeviceName");
                 if (vid_frmsize.discrete.width == 1024 && vid_frmsize.discrete.height == 768){
                     mSizeFPSParamIdx ++;
                     vid_frmsize.index = mSizeFPSParamIdx;
@@ -199,7 +199,7 @@ namespace android{
             pCapcfg->tv.numerator = 1;
             pCapcfg->tv.denominator = 30;
         }
-        LOGD("the fps is %d", pCapcfg->tv.denominator);
+        CAMERA_HAL_LOG_RUNTIME("the fps is %d", pCapcfg->tv.denominator);
 
         parm.parm.capture.timeperframe.numerator = pCapcfg->tv.numerator;
         parm.parm.capture.timeperframe.denominator = pCapcfg->tv.denominator;
@@ -255,7 +255,6 @@ namespace android{
             CAMERA_HAL_LOG_RUNTIME(" pixelformat = %x\n", fmt.fmt.pix.pixelformat);
         }
         pCapcfg->framesize = fmt.fmt.pix.sizeimage;
-        pCapcfg->picture_waite_number = 1; //For uvc, the first frame is ok.
 
         return CAPTURE_DEVICE_ERR_NONE;
     }
@@ -273,7 +272,7 @@ namespace android{
         unsigned int pic_waite_buf_num = 0;
         if ((strstr(mInitalDeviceName, OV5640_NAME_STR) != 0) ||
                 (strstr(mInitalDeviceName, OV5642_NAME_STR) != 0)){
-            pic_waite_buf_num = 2;
+            pic_waite_buf_num = 10;
             if (capturewidth == 640 && captureheight == 480)
                 capturemode = 0;	/* VGA mode */
             else if (capturewidth == 320 && captureheight == 240)
@@ -295,7 +294,7 @@ namespace android{
                 return CAPTURE_DEVICE_ERR_BAD_PARAM;
             }
         }else if(strstr(mInitalDeviceName, OV3640_NAME_STR) != 0){
-            pic_waite_buf_num = 2;
+            pic_waite_buf_num = 10;
             if (capturewidth == 320 && captureheight == 240)
                 capturemode = 1;	/* QVGA mode */
             else if (capturewidth == 640 && captureheight == 480)
-- 
1.8.0

